src/mlx/warnings.py

```diff
@@ -148,32 +154,58 @@ def toggle_printout(self, printout):
         '''
         self.printout = printout
 
+    def config_parser(self, config):
```

Comment:
this name might be confusing (see https://docs.python.org/3/library/configparser.html which parses INI files)

can we rename e.g. config_from_json() or other?